server/sonar-server/src/test/java/org/sonar/server/test/index/TestIndexTest.java

```diff
@@ -0,0 +1,111 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.test.index;
+
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.sonar.api.config.Settings;
+import org.sonar.server.es.EsTester;
+
+import java.util.*;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestIndexTest {
+  @ClassRule
+  public static EsTester es = new EsTester().addDefinitions(new TestIndexDefinition(new Settings()));
+
+  TestIndex sut = new TestIndex(es.client());
+
+  @Before
+  public void before() throws Exception {
+    es.truncateIndices();
+  }
+
+  @Test
+  public void lines_covered_a_test_method() throws Exception {
+    List<Map<String, Object>> coverageBlocks = new ArrayList<>();
+    coverageBlocks.add(newCoverageBlock("1"));
+    coverageBlocks.add(newCoverageBlock("2"));
+    coverageBlocks.add(newCoverageBlock("3"));
+
+    es.putDocuments(TestIndexDefinition.INDEX, TestIndexDefinition.TYPE,
+      newTestDoc("1", newCoverageBlock("3"), newCoverageBlock("4"), newCoverageBlock("5")),
+      newTestDoc("2", newCoverageBlock("5"), newCoverageBlock("6"), newCoverageBlock("7")));
+
+    List<Map<String, Object>> result = sut.coveredLines("uuid-1", "name-1");
+
+    assertThat(result).hasSize(3);
+  }
+
+  @Test
+  public void test_methods() throws Exception {
+    List<Map<String, Object>> coverageBlocks = new ArrayList<>();
+    coverageBlocks.add(newCoverageBlock("1"));
+    coverageBlocks.add(newCoverageBlock("2"));
+    coverageBlocks.add(newCoverageBlock("3"));
+
+    es.putDocuments(TestIndexDefinition.INDEX, TestIndexDefinition.TYPE,
+      newTestDoc("1", newCoverageBlock("3"), newCoverageBlock("4"), newCoverageBlock("5")),
+      newTestDoc("1", newCoverageBlock("5"), newCoverageBlock("6"), newCoverageBlock("7")));
+
+    List<TestDoc> result = sut.testMethods("uuid-1");
+
+    assertThat(result).hasSize(2);
+  }
+
+  @Test
+  public void test_covering() throws Exception {
+    List<Map<String, Object>> coverageBlocks = new ArrayList<>();
+    coverageBlocks.add(newCoverageBlock("1"));
+    coverageBlocks.add(newCoverageBlock("2"));
+    coverageBlocks.add(newCoverageBlock("3"));
+
+    es.putDocuments(TestIndexDefinition.INDEX, TestIndexDefinition.TYPE,
+      newTestDoc("1", newCoverageBlock("3"), newCoverageBlock("4"), newCoverageBlock("5")),
+      newTestDoc("1", newCoverageBlock("5"), newCoverageBlock("6"), newCoverageBlock("7")));
+
+    List<TestDoc> result = sut.testsCovering("main-uuid-5", 82);
+
+    assertThat(result).hasSize(2);
+  }
+
+  private Map<String, Object> newCoverageBlock(String id) {
+    Map<String, Object> coverageBlock = new HashMap<>();
+    coverageBlock.put("key", "project:file-" + id);
+    coverageBlock.put("uuid", "main-uuid-"+ Integer.valueOf(id));
+    coverageBlock.put("coveredLines", 42);
```

Comment:
If my understanding is correct, coveredLines is the number of line covered by the test. As you've added only 3 lines in the 'lines'  parameter, you should set 3 in 'coveredLines'.
Do not forget that tests are also used to understand what is doing the class !